common/persistence/visibility/store/query/interceptors.go

```diff
@@ -0,0 +1,13 @@
+package query
+
+type SearchAttributeInterceptor interface {
```

Comment:
Suggest a comment on the semantics of this interceptor, and what is it used for.